### PR TITLE
[MB-12729] delete customer support remarks

### PR DESCRIPTION
**Pull request by mboren**
## Summary

I implemented the backend endpoint for deleting customer support remarks and wired it up to the existing `delete` button on customer support remarks.

I added a `deleted_at` column to `customer_support_remarks` to allow soft deletion.

I also created a confirmation modal that appears when you click the delete button, and a success notification that appears after a remark has successfully been deleted.

I did not add any authorization/permissions stuff to the new endpoint because we're still figuring out what pattern to use on the backend for situations where a user only has permissions to do something to objects that they created. (see [MB-12730](https://dp3.atlassian.net/browse/MB-12730))

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Login to the office app as a QAE/CSR user
2. Search for a move and click on it to go to the move details page
3. Go to the "Customer support remarks" tab
4. Add a few remarks
5. Verify that you can delete the remarks with the delete button
6. Verify that a green success notification appears after you delete a remark
7. Create a new QAE/CSR user, navigate to the same move, and verify that you cannot delete remarks created by your original user.

Storybook link for confirmation modal: http://localhost:6006/?path=/story/components-deletecustomersupportremarkconfirmationmodal--basic

## Verification Steps for Author

These are to be checked by the author.

- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database

## Screenshots

https://user-images.githubusercontent.com/2627844/176319810-b8ed5ed2-baae-4790-b8ea-18178220c1f6.mov

### CODE REVIEW

**Review by leia-sefkin — APPROVED** [review]
Reviewed and pulled down code locally, everything looks good and is running well. Good test coverage here as well

File: pkg/services/customer_support_remarks/customer_support_remarks_deleter.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+type customerSupportRemarkDeleter struct {
+}
+
+func NewCustomerSupportRemarkDeleter() services.CustomerSupportRemarkDeleter {
+	return &customerSupportRemarkDeleter{}
+}
+
+func (o customerSupportRemarkDeleter) DeleteCustomerSupportRemark(appCtx appcontext.AppContext, customerSupportRemarkID uuid.UUID) error {
+	var remark models.CustomerSupportRemark
+	err := appCtx.DB().Q().Scope(utilities.ExcludeDeletedScope()).Find(&remark, customerSupportRemarkID)
```
Comment (duncan-truss) [inline pkg/services/customer_support_remarks/customer_support_remarks_deleter.go:26]: 🙌 Thanks for incorporating the scope.  You may not need the Q.() part if both Scope and Find works on the connection.

**Review by LeDeep — APPROVED** [review]
approving db changes

File: pkg/services/customer_support_remarks/customer_support_remarks_deleter.go (hunk 1 above)
Comment (duncan-truss) [inline pkg/services/customer_support_remarks/customer_support_remarks_deleter.go:26]: 🙌 Thanks for using the scope

**Review by cosburn — APPROVED** [review]
Looks great!

File: pkg/services/customer_support_remarks/customer_support_remarks_deleter.go (hunk 1 above)
Comment (mboren) [reply pkg/services/customer_support_remarks/customer_support_remarks_deleter.go]: You're right I don't need the `.Q()`, thanks!

**Pull request merged**